### ISSUE TITLE
[BUG] Refactor GetAlerts API to not throw 404 NOT_FOUND errors

### ISSUE THREAD

**Issue by AWSHurneyt** [opensearch-project/security-analytics#872]
**What is the bug?**
The GetAlerts API throws a `404 NOT_FOUND` error when there are no detectors for the specified `detectorType`.
https://github.com/opensearch-project/security-analytics/blob/main/src/main/java/org/opensearch/securityanalytics/transport/TransportGetAlertsAction.java#L126

This can cause a built-up of `4xx` errors on clusters that make regular/repeated calls to that API.

**What is the expected behavior?**
Rather than throw an error, the GetAlerts API could be refactored to instead return an empty list when there are no detectors defined for the specified `detectorType`.

**Comment by AWSHurneyt** [comment]
@shanjare2002 
Sure! Feel free to raise a PR.

**Comment by eirsep** [comment]
@shanjare2002 are you fixing this?

**Comment by ishag4** [comment]
Hi @AWSHurneyt @eirsep If this is still pending. I would like to contribute, you can assign me.

**Comment by AWSHurneyt** [comment]
Hi @CodeByTJ15! 

Yes, this issue is still open. 

To test out whether the issue is fixed, you can run the GetAlerts API ([doc link](https://docs.opensearch.org/docs/latest/security-analytics/api-tools/alert-finding-api/#example-request)) using a `detectorType` that doesn't have any detectors created. For example, if I have 0 windows log type detectors configured (or if I just have 0 detectors configured all together), this API call through devtools will result in this error.
```
GET /_plugins/_security_analytics/alerts?detectorType=windows

{
  "error": {
    "root_cause": [
      {
        "type": "security_analytics_exception",
        "reason": "No detectors found for provided type"
      }
    ],
    "type": "security_analytics_exception",
    "reason": "No detectors found for provided type",
    "caused_by": {
      "type": "exception",
      "reason": "org.opensearch.OpenSearchStatusException: No detectors found for provided type"
    }
  },
  "status": 404
}
```

**Comment by CodeByTJ15** [comment]
Hi @AWSHurneyt ,
Can you let me know if below is expected?
``
{
    "alerts": [],
    "total_alerts": 0
}
``

**Comment by AWSHurneyt** [comment]
@CodeByTJ15 yes, a response with an empty `alerts` array, and 0 `total_alerts` should be fine. Just as a heads up, there may be some unit/integration tests that will need refactoring to reflect that change.

**Comment by CodeByTJ15** [comment]
Hey @AWSHurneyt ,
thanks for replying. I'm unable to find existing tests for empty array list. can you please guide on this?

**Comment by AWSHurneyt** [comment]
@CodeByTJ15 Here's one such test. There may be others as well. However, once a PR is published and the github actions start running, we should see what other tests are impacted.
https://github.com/opensearch-project/security-analytics/blob/main/src/test/java/org/opensearch/securityanalytics/alerts/AlertsIT.java#L295
